tests/src/Test/Tasty/Clash.hs

```diff
@@ -476,34 +457,33 @@ runTest1 modName opts@TestOptions{..} path target =
   --
   -- where groups B and C are independent of each other, but both dependent on
   -- the success of Task A1.
-
-  vivado tmpDir =
-    case target of
-      SystemVerilog ->
-        -- TODO: Get SystemVerilog working on Vivado
-        []
-      _ -> buildAndSimTests Vivado (vivadoTests opts tmpDir modName)
-
-  nonVerilator tmpDir =
-    case target of
-      VHDL -> buildAndSimTests GHDL (vhdlTests opts tmpDir)
-      Verilog ->
-        case verilate of
-          VerilateOnly -> []
-          _ -> buildAndSimTests ModelSim (verilogTests opts tmpDir)
-
-      SystemVerilog ->
-        case verilate of
-          VerilateOnly -> []
-          _ -> buildAndSimTests ModelSim (systemVerilogTests opts tmpDir)
-
-  verilator tmpDir =
-    case target of
-      VHDL -> []
-      _ ->
-        case verilate of
-          SimOnly -> []
-          _ -> buildAndSimTests Verilator (verilatorTests opts tmpDir)
+  sequenceIndependent
+    :: (TestName, TestTree)
+    -- ^ The tree everything else depends on ("clash (gen)")
+    -> [[(TestName, TestTree)]]
+    -- ^ All the independent trees to run after "clash (gen)"
+    -> [TestTree]
+  sequenceIndependent gen = go
+   where
+    go [] = sequenceOne []
+    go (l:ls) = sequenceOne l <> concatMap (tail . sequenceOne) ls
+    sequenceOne ts = sequenceTests (show target : path) (gen : ts)
+
+  -- | The tests that are switched by `hdlLoad` and `hdlSim`
+  hdlTests tmpDir = case target of
+    VHDL ->
+      [ buildAndSimTests GHDL (ghdlTests opts tmpDir)
+      , buildAndSimTests Vivado (vivadoTests opts tmpDir modName)
+      ]
+    Verilog ->
+      [ buildAndSimTests IVerilog (iverilogTests opts tmpDir)
+      , buildAndSimTests Verilator (verilatorTests opts tmpDir)
+      , buildAndSimTests Vivado (vivadoTests opts tmpDir modName)
+      ]
+    SystemVerilog ->
+      [ buildAndSimTests ModelSim (modelsimTests opts tmpDir)
+      , buildAndSimTests Verilator (verilatorTests opts tmpDir)
+      ]
```

Comment:
ModelSim should be able to do Verilog and VHDL too. Maybe add a TODO?

Reply:
Done